thunder/torch/experimental/dtensor_proxy.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional
+
+from thunder.core.proxies import TensorProxy, AnyProxy, _infer_tensor_properties
+from torch.distributed._tensor import DTensor
+from thunder.core.proxies import proxy
+import thunder.core.devices as devices
+import thunder.core.dtypes as dtypes
+
+
+# Inherit from TensorProxy as DTensor also supports
+# Tensor methods like __add__, __div__, sin, etc.
```

Comment:
As I don't remember the behavior, would `DTensorProxy.__add__` and others return an instance of `DTensorProxy` or `TensorProxy`?

Reply:
`DTensorProxy.__add__` will return an instance of DTensorProxy as after the method resolution, it will finally dispatch to the DTensor symbol.

For a method which hasn't been implemented yet for DTensor, it will error out with 
```
Expected all inputs to be TensorProxy but found {list(map(lambda t: type(t), filter_tensor_proxies))}
```